tests/test_api.py

```diff
@@ -294,3 +297,61 @@ def test_url_protocol_association(delete_files):
         url_to_open=url,
         expected_output=url,
     )
+
+
+@pytest.mark.skipif(PLATFORM != "win", reason="Windows only")
+def test_windows_terminal_profiles():
+    def _parse_terminal_profiles(settings_file: Path) -> dict:
+        settings = json.loads(settings_file.read_text())
+        return {
+            profile.get("name", ""): profile.get("commandline", "")
+            for profile in settings["profiles"]["list"]
+        }
+
+    settings_files = [
+        # Stable
+        Path(
+            folder_path("user", False, "localappdata"),
+            "Packages",
+            "Microsoft.WindowsTerminal_8wekyb3d8bbwe",
+            "LocalState",
+            "settings.json",
+        ),
+        # Preview
+        Path(
+            folder_path("user", False, "localappdata"),
+            "Packages",
+            "Microsoft.WindowsTerminalPreview_8wekyb3d8bbwe",
+            "LocalState",
+            "settings.json",
+        ),
+        # Unpackaged (Scoop, Chocolatey, etc.)
+        Path(
+            folder_path("user", False, "localappdata"),
+            "Microsoft",
+            "Windows Terminal",
+            "settings.json",
+        ),
+    ]
+    settings_files = [file for file in settings_files if file.exists()]
+    if not settings_files:
+        pytest.skip("No terminal profile settings file found.")
```

Comment:
Is Terminal preinstalled on GHA CI? 🤔 I don't see it in https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md

Reply:
Well, that led me down a rabbit hole...

I discovered that Terminal does not have the config files set up after installation. You have to run it to write the files. So, I had to change some of the behavior of how the files are detected. Luckily, writing the a file and opening the terminal does not remove the profiles, so we're good there.

The second problem was that `folder_path` is mocked away in the tests. So, I had to change the tests to create the files inside the mocked directory. Additionally, I had to add a custom mark to use the "true" paths on the CI.

However, I discovered that using chocolatey to install the terminal onto the GitHub runner actually doesn't work even though the installation succeeded: https://github.com/conda/menuinst/actions/runs/9023438085/job/24795241900#step:4:2662

`wt` should have been available, so I have no idea why the CI behaves so differently than a local system. I'm a little stumped here.